lib/XMLHttpRequest.js

```diff
@@ -490,7 +524,7 @@ exports.XMLHttpRequest = function () {
         response = result.data
         self.status = result.data.statusCode
         if (encoding === 'binary') {
-          self.response = Uint8Array.from(result.data.binary.data).buffer
```

Comment:
This conversion was unnecessary as a Buffer is a Uint8Array.

Reply:
Reading the code again, I don't even remember how `data` is defined. I can see that we assigned the Buffer to `result.data.binary` but I don't explain the last `.data`.

Anyway I think you're right and the code can be simplified.

Reply:
`result.data.binary` is stringify so `result.data.binary.data` is an `Array` not a `Buffer` or an `Uint8Array`.
The specification says that it should return a `ArrayBuffer`.

Reply:
Fixed (after a lot of learning) in a new commit.

Reply:
> Fixed (after a lot of learning) in a new commit.

:+1: 
I think we should add a comment to preserve this knowledge. We should explain why we are using `new Uint8Array()` instead `Uint8Array.from` (or if we could use one or the other).
And we should also state the type of `result.data.binary`: "result.data.binary is stringify so result.data.binary.data is an Array"